fdbclient/GlobalConfig.actor.cpp

```diff
@@ -155,45 +155,56 @@ void GlobalConfig::erase(KeyRangeRef range) {
 // function performs a one-time migration of data in these keys to the new
 // global configuration key space.
 ACTOR Future<Void> GlobalConfig::migrate(GlobalConfig* self) {
-	state Reference<ReadYourWritesTransaction> tr = makeReference<ReadYourWritesTransaction>(self->cx);
-	tr->setOption(FDBTransactionOptions::ACCESS_SYSTEM_KEYS);
-
-	state Key migratedKey("\xff\x02/fdbClientInfo/migrated/"_sr);
-	state Optional<Value> migrated = wait(tr->get(migratedKey));
-	if (migrated.present()) {
-		// Already performed migration.
-		return Void();
-	}
+	state Reference<ReadYourWritesTransaction> tr;
+	try {
+		loop {
+			tr = makeReference<ReadYourWritesTransaction>(Database(Reference<DatabaseContext>::addRef(self->cx)));
+			wait(delay(0));
```

Comment:
Based on https://github.com/apple/foundationdb/pull/7127#discussion_r870519147. Is it not necessary in this case?

Reply:
I see, so this is to ensure that the `DatabaseContext` can be dereferenced. In that case, I think the `delay` needs to happen while we aren't holding a reference, so we need to:

1. Remove our reference to it (e.g. by clearing out the transaction)
2. Do the delay
3. Create a new transaction. 

Possibly this could all be a procedure done in the catch block when we are retrying, though it probably doesn't hurt much to do it at the top either.

Reply:
I added an implementation of this, I think the `tr->onError` is still necessary, but when it's followed directly by `tr.clear()` it feels a little strange.